### ISSUE TITLE
Prefer server side copy on Google Drive

### ISSUE THREAD

**Issue by anyone00** [rclone/rclone#4696]
Btw needless to say I can't thank you enough for creating/maintaining this awesome project ♥

#### What is your current rclone version (output from `rclone version`)?

rclone v1.53.1
- os/arch: linux/amd64
- go version: go1.15

#### What problem are you are trying to solve?

copy on server side to save bandwidth

now it is really confusing imo to figure out if any of the following server-side operations are supported by:

Team Drive to My Drive
Team Drive to Team Drive
My Drive to Team Drive
My Drive to My Drive

the same goes for enterprise gsuite members/free gsuite accounts (schools)...etc

#### How do you think rclone should be changed to solve that?

basically default to server-side copying, if not applicable switch to download/upload method

- First rclone will check if src and dst configurations support server-side operation
- If both support server-side operations it will proceed by doing the server-side operation
- Otherwise fallback to download then upload.

**Comment by anyone00** [comment]
This is also related to #3119

**Comment by Skylion007** [comment]
This used to be the default, if I recall. 

Unfortunately, Google implemented a 100GB Server-Side copy limit (per user per day) which is much lower than the 750GB daily upload limit. 

Therefore, it often makes sense to just default to client side copy.

There is a flag to enable Google Drive server side copy though. Feel free to use that.

**Comment by anyone00** [comment]
Thanks I did not know these was a 100GB limit on server side copies. Is it documented somewhere?
